sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/StorageInputStream.java

```diff
@@ -0,0 +1,391 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.common;
+
+import com.azure.core.util.logging.ClientLogger;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+
+/**
+ * Provides an input stream to read a given storage resource.
+ */
+public abstract class StorageInputStream extends InputStream {
+    private final ClientLogger logger = new ClientLogger(StorageInputStream.class);
+
+    /**
+     * A flag to determine if the stream is faulted, if so the last error will be thrown on next operation.
+     */
+    private volatile boolean streamFaulted;
```

Comment:
>privat [](start = 4, length = 6)

I think this should be protected. I don't see a need for the parent and child types to have their own streamFaulted fields.

Reply:
Done. Good point